shell/common/asar/archive_mac.mm

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2021 Slack Technologies, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/common/asar/archive.h"
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <Foundation/Foundation.h>
+
+#include <string>
+
+#include "base/logging.h"
+#include "base/mac/foundation_util.h"
+#include "base/strings/sys_string_conversions.h"
+#include "electron/fuses.h"
+
+namespace asar {
+
+void Archive::MaybeValidateArchiveSignature() {
+// This method uses a private API SecCodeValidateFileResource
+// which should not be used in the MAS build
+#ifndef MAS_BUILD
+  // If the fuse is not enabled, don't waste any time here.
+  if (!electron::fuses::IsEmbeddedAsarSignatureValidationEnabled())
+    return;
+
+  NSURL* bundle_url =
+      [NSURL URLWithString:[[NSBundle mainBundle] executablePath]];
+  // /Stuff/Electron.app/Contents/MacOS/Electron is the bundle url.
+  NSURL* contents_url = [[bundle_url URLByDeletingLastPathComponent]
+      URLByDeletingLastPathComponent];
+  const std::string abs_path = path_.value();
+
+  // If the path to the app.asar file is not contained by the bundle we can't
+  // validate the signature.
+  if (abs_path.rfind(base::SysNSStringToUTF8([contents_url absoluteString]),
```

Comment:
Is this the right way to check contained-ness? Does this work in the face of symlinks?

e.g. if `/a` is symlinked to `/b`, then `/a/foo/app.asar` _is_ contained within `/b/foo`, but this check will fail.

Reply:
A file being symlinked from inside to the bundle to outside the bundle isn't possible I don't think 🤔  But regardless the path passed in to the asar archive is absolute and fully resolved, as is the `NSBundle` url.  There is no possibility for a symlink in the way given how we resolve paths and how we provide `process.resourcesPath`

Reply:
Maybe we could use [base::FilePath::IsParent](https://source.chromium.org/chromium/chromium/src/+/main:base/files/file_path.h;l=227-233;drc=665c2f724a8e64c3253010100bdb77ed6bb39632) tho?

Reply:
oh, macOS filesystems are case-insensitive by default, too, so it's totally possible to have two paths that do not string-match but one contains the other.